software_package/in4073/in4073.c

```diff
@@ -77,24 +101,32 @@ int main(void)
 
 	uint32_t counter = 0;
 	demo_done = false;
+	mode = 0;
 
 	while (!demo_done)
 	{
-		if (rx_queue.count) process_key( dequeue(&rx_queue) );
+		if (rx_queue.count) 

```

Comment:
Will we not loose out on reading some data here?, because the queue is being dequeued earlier by process_key function.

Reply:
I don't see where you see a problem. Process_key doesn't dequeue anything. Only one byte is dequeued and process_packet is called for that byte. The read byte will be stored in the array in the corresponding file (packet.c). The rest will continue as usual.

Reply:
Maybe I'm reading it wrong. The process_key and process_packet function never act on the same element of the queue, right?